src/index/mod.rs

```diff
@@ -155,6 +155,10 @@ impl IndexRecord {
 
 pub trait Fields {
     fn get_field_values(&self) -> Vec<(&'static str, u64)>;
+
+    fn get_record_size(&self) -> u64 {
```

Comment:
I don't think this should be u64, it is at most a usize or i32 based on how it is being used above